pkg/app/api/service/webservice/service.proto

```diff
@@ -442,15 +444,25 @@ message ListAPIKeysResponse {
     repeated model.APIKey keys = 1;
 }
 
-message GetInsightDataRequest {
+message GetInsightMetricsDataRequest {
     pipe.model.InsightMetricsKind metrics_kind = 1 [(validate.rules).enum.defined_only = true];
     pipe.model.InsightStep step = 2 [(validate.rules).enum.defined_only = true];
     int64 range_from = 3 [(validate.rules).int64.gt = 0];
     int64 data_point_count = 4 [(validate.rules).int64.gt = 0];
     string application_id = 5;
 }
 
-message GetInsightDataResponse {
+message GetInsightDataPointsResponse {
     int64 updated_at = 1;
     repeated pipe.model.InsightDataPoint data_points = 2;
 }
+
+message GetInsightPieChartDataRequest {
```

Comment:
I think instead of adding a new RPC for this case, it would be nice if `GetInsightData` can be abstracted for all cases.
And `Pie` does not need to be included in the name because the API should not depend on how the data is rendered.
The client retrieves the data from API and renders that data in their way, they can use `pie`, `histograms`, `line`, or anything else.

So I think we can learn from the definition of Prometheus
https://github.com/prometheus/common/blob/317b7b125e8fddda956d0c9574e5f03f438ed5bc/model/value.go

``` proto
// model/insight.proto

enum InsightResultType {
  MATRIX = 1;
  VECTOR = 2;
}

message InsightDataPoint {
  int64 timestamp = 1 [(validate.rules).int64.gt = 0];
  float value = 2 [(validate.rules).float.gt = 0];
}

message InsightSample {
   map<string,string> labels = 1;
   InsightDataPoint data_point = 2;
}

message InsightSampleStream {
  map<string,string> labels = 1;
  repeated InsightDataPoint data_points = 3;
}

// webservice/service.proto

message GetInsightDataResponse {
     int64 udpated_at = 1;
     repeated pipe.model.InsightDataPoint data_points = 2 [deprecated = true];
     type model.InsightResultType = 3;
     repeated model.InsightSample vector = 4;
     repeated model.InsightSampleStream matrix = 5;
}
```

In the case of Deployment Insight, the response looks like

``` golang
GetInsightDataResponse {
  Type: MATRIX,
  Matrix: [{
    Labels: nil,
    DataPoints: {...},
  }],
},
```

In the case of Application Insight, the response looks like

``` golang
GetInsightDataResponse {
  Type: VECTOR,
  Vector: [
    {
      Labels:{
        "AppKind": "KUBERNETES",
        "Deleted": "true",
        "EnableStatus": "enable",
      },
      DataPoint: {},
    },
    {
      Labels:{
        "AppKind": "TERRAFORM",
        "Deleted": "false",
        "EnableStatus": "enable",
      },
      DataPoint: {},
    },
},
```

@sanposhiho Could you tell me your thoughts?

Reply:
I'm sorry for the late reply. (Please complain to my teacher who gave me a huge amount of term papers 😠)

> I think instead of adding a new RPC for this case, it would be nice if GetInsightData can be abstracted for all cases.
And Pie does not need to be included in the name because the API should not depend on how the data is rendered.

Agree :D 
I hadn't thought of the above way that use `GetInsightDataResponse` to change its contents depending on the shape of the graph.

I'll fix it based on the above way, thanks 👍

Reply:
@sanposhiho Thank you for your great contributions.

> Please complain to my teacher who gave me a huge amount of term papers.

Haha. Please send me your teacher's email. I can send a message to him. (just a joke 😄 )

> I'll fix it based on the above way, thanks 👍

Nice. Thank you.